src/example/example.py

```diff
@@ -21,51 +29,76 @@
 # Third-Party Libraries
 import docopt
 import pkg_resources
+from schema import And, Schema, SchemaError, Use
 
 from ._version import __version__
 
 DEFAULT_ECHO_MESSAGE = "Hello World from the example default!"
 
 
-def example_div(x, y):
+def example_div(dividend, divisor):
     """Print some logging messages."""
     logging.debug("This is a debug message")
     logging.info("This is an info message")
     logging.warning("This is a warning message")
     logging.error("This is an error message")
     logging.critical("This is a critical message")
-    return x / y
+    return dividend / divisor
 
 
 def main():
     """Set up logging and call the example function."""
     args = docopt.docopt(__doc__, version=__version__)
-    # Set up logging
-    log_level = args["--log-level"]
+    # Validate and convert arguments as needed
+    schema = Schema(
+        {
+            "--log-level": And(
+                str,
+                Use(str.lower),
+                lambda n: n in ("debug", "info", "warning", "error", "critical"),
+                error="Possible values for --log-level are "
+                + "debug, info, warning, error, and critical.",
+            ),
+            "<dividend>": Use(int, error="<dividend> must be an integer."),
+            "<divisor>": And(
+                Use(int),
+                lambda n: n != 0,
+                error="<divisor> must be an integer that is not 0.",
+            ),
+            str: object,  # Don't care about other keys, if any
+        }
+    )
+
     try:
-        logging.basicConfig(
-            format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
-        )
-    except ValueError:
-        logging.critical(
-            f'"{log_level}" is not a valid logging level.  Possible values '
-            "are debug, info, warning, and error."
-        )
+        args = schema.validate(args)
+    except SchemaError as err:
+        # Exit because one or more of the arguments were invalid
+        print(err, file=sys.stderr)
```

Comment:
This could be logged to `critical` instead of simply being printed to standard error.

Reply:
Well this is a little different than the other output.  This is the error message describing why you messed up the arguments. We could do that, but I think it is a bit weird.  At this point you're dealing with a user that is pretty clueless.  Since the logger isn't configured yet it makes the error message less clear:

This is how it is now:
```console
skeleton-python-library ❱ example 1 0
<divisor> must be an integer that is not 0.
``` 

If this error was printed through the logging system:
```console
skeleton-python-library ❱ example 1 0
CRITICAL:root:<divisor> must be an integer that is not 0.
```

I would prefer not to change it, but I will if the team thinks otherwise.

Here is non-error output to see what a configured logger looks like:

```console
skeleton-python-library ❱ example 1 1
2020-01-29 23:18:49,609 INFO This is an info message
2020-01-29 23:18:49,609 WARNING This is a warning message
2020-01-29 23:18:49,609 ERROR This is an error message
2020-01-29 23:18:49,609 CRITICAL This is a critical message
2020-01-29 23:18:49,609 INFO 1 / 1 == 1.0
2020-01-29 23:18:49,610 INFO ECHO_MESSAGE="Hello World from the example default!"
2020-01-29 23:18:49,612 INFO Secret="Three may keep a secret, if two of them are dead."
```

Reply:
One other thought I had was this output is similar to what `docopt` emits for `--help`, or invalid syntax.  It would be a bit absurd to prepend its output with the same logging cruft:

```console
skeleton-python-library ❱ example 0
Usage:
  example [--log-level=LEVEL] <dividend> <divisor>
  example (-h | --help)
```

```console
skeleton-python-library ❱ example 0
CRITICAL:root:Usage:
CRITICAL:root:  example [--log-level=LEVEL] <dividend> <divisor>
CRITICAL:root:  example (-h | --help)
```

Reply:
Those are good points @felddy - you have convinced me that `print` makes more sense here than the `logger`.

Reply:
You are :100: percent correct, @felddy.  Please disregard my comment.